qemu/tests/live_backup_base.py

```diff
@@ -0,0 +1,213 @@
+import time
+import logging
+
+from autotest.client.shared import error
+from virttest import utils_misc
+from virttest import qemu_storage
+
+from qemu.tests import block_copy
+
+
+@error.context_aware
+class LiveBackup(block_copy.BlockCopy):
+
+    """
+    Provide basic functions for live backup test cases.
+    """
+
+    def __init__(self, test, params, env, tag):
+        """
+        Init the default values for live backup object.
+
+        :param test: Kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        :param tag: Image tag defined in parameter images
+        """
+        block_copy.BlockCopy.__init__(self, test, params, env, tag)
+        self.image_chain = self.params.get("image_chain").split()
+        self.image_cmd = utils_misc.get_qemu_img_binary(params)
+        self.source_image = self.params.get("source_image")
+        self.speed = int(self.params.get("speed", 0))
+        self.bitmap_name = "bitmap0"
+        self.backup_params = self.params.object_params("source_image")
+        self.backup_index = 1
+        self.backup_format = self.params.get("backup_format")
+        self.generate_backup_params()
+
+    def generate_backup_params(self):
+        """
+        Generate params for full backup image.
+        """
+        backup_params = self.params.object_params("source_image")
```

Comment:
backup_params = self.backup_params
which has already been initialized in line 33.